### DISCUSSION TITLE
Custom Plugin - is last cell logic

### DISCUSSION THREAD

**Discussion by twoxfh** [gzuidhof/starboard-notebook#102] [Q&A] [answered]
In a Custom Plugin is there any way to identify the last cell and only run additional logic on the last cell? I have a SQLCell plugin but I only want to close the database on the last cell. 

Thanks for reading.

**Comment by unhott** [comment, marked as answer]
I *think* that should be technically straightforward. 
ES Module cell: 
```
export const plugin = {
  id: "test",
  metadata: {
    name: "test",
  },
  exports: {},
  async register(runtime, opts) {
    runtime.dom.notebook.addEventListener("sb:run_cell", (event) => {
      let currentCellId = event.target.id;
      
      let rt = event.target.runtime;
      let cells = rt.dom.cells;
      let lastCellId = cells[cells.length-1].id;
      let isLastCell = currentCellId == lastCellId;
      console.log(isLastCell);
    });
  }
};

runtime.controls.registerPlugin(plugin)
```

I am not confident if this is the right approach for your problem, though! 
Good luck :)

**Reply by twoxfh** [reply]
Thank you, this works as a prehook to a cell, any idea about a post hook, where the cell runs then executes the last bit if code? Will think more about the approach, but this works in identifying the last cell.

**Reply by unhott** [reply]
Ah, I see the problem. Maybe you can define a cell property to identify the final SQL cell in a notebook. Listen to events like move / insert cell etc to update the property automatically. Then modify the run behavior to check if this property exists on the cell. 

Or maybe define a new cell type that terminates this connection when run?

**Reply by twoxfh** [reply]
Thanks for your feedback. I am using an if statement and watching for it in the run logic, maybe in the future there will be a more elegant way.

**Reply by unhott** [reply]
I think I realized later that a property would be unnecessary, It sounds like you figured that out ;)
